aws/internal/service/elasticache/finder/finder.go

```diff
@@ -125,3 +125,57 @@ func CacheClustersByID(conn *elasticache.ElastiCache, idList []string) ([]*elast
 
 	return results, err
 }
+
+// GlobalReplicationGroupByID() retrieves an ElastiCache Global Replication Group by id.
+func GlobalReplicationGroupByID(conn *elasticache.ElastiCache, id string) (*elasticache.GlobalReplicationGroup, error) {
+	input := &elasticache.DescribeGlobalReplicationGroupsInput{
+		GlobalReplicationGroupId: aws.String(id),
+		ShowMemberInfo:           aws.Bool(true),
+	}
+	output, err := conn.DescribeGlobalReplicationGroups(input)
+	if tfawserr.ErrCodeEquals(err, elasticache.ErrCodeGlobalReplicationGroupNotFoundFault) {
+		return nil, &resource.NotFoundError{
+			LastError:   err,
+			LastRequest: input,
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if output == nil || len(output.GlobalReplicationGroups) == 0 || output.GlobalReplicationGroups[0] == nil {
+		return nil, &resource.NotFoundError{
+			Message:     "empty result",
+			LastRequest: input,
+		}
+	}
+
+	return output.GlobalReplicationGroups[0], nil
+}
+
+// GlobalReplicationGroupMemberByID retrieves a member Replication Group by id from a Global Replication Group.
+func GlobalReplicationGroupMemberByID(conn *elasticache.ElastiCache, globalReplicationGroupID string, id string) (*elasticache.GlobalReplicationGroupMember, error) {
+	globalReplicationGroup, err := GlobalReplicationGroupByID(conn, globalReplicationGroupID)
+	if err != nil {
+		return nil, &resource.NotFoundError{
+			Message:   "unable to retrieve enclosing Global Replication Group",
+			LastError: err,
+		}
+	}
+
+	if len(globalReplicationGroup.Members) == 0 {
```

Comment:
To prevent potential panics should the prior code change, should double check that `globalReplicationGroup` is not `nil` here as well:

```suggestion
	if globalReplicationGroup == nil || len(globalReplicationGroup.Members) == 0 {
```

Although the length check itself is extraneous as the `for` loop will not match the result and the bottom error will be returned.